pvlib/losses.py

```diff
@@ -89,14 +115,26 @@ def soiling_hsu(rainfall, cleaning_threshold, tilt, pm2_5, pm10,
     return soiling_ratio
 
 
+def _fix_tmy_monotonicity(rain_index, rain_vals, timestep_interval, rain_tz,
+                          rain_name):
+    # fix TMY to be monotonically increasing by rolling index by 1 interval
```

Comment:
General question: what is the reasoning for replacing the time index here for soiling functions, rather than incrementing the year for the last time step in `read_tmy3(<tmy file>, coerce_year=<year>)`?

Reply:
Because it will change the existing, expected API for `read_tmy3()` so I thought it deserved a bigger discussion. I think your comment is similar to Will's above. I'm okay with either approach, although I have a preference for breaking the API and changing `read_tmy3()` to be monotonic when coercing the year. All in favor?

Reply:
I'm in favor, but should it be done in 0.8 instead of 0.7.2?